libyang/context.py

```diff
@@ -424,7 +440,33 @@ def parse_data_file(
             fmt,
             in_type=IOType.FD,
             in_data=fileobj,
-            parent=parent,
+            parser_lyb_mod_update=parser_lyb_mod_update,
+            parser_no_state=parser_no_state,
+            parser_parse_only=parser_parse_only,
+            parser_opaq=parser_opaq,
+            parser_ordered=parser_ordered,
+            parser_strict=parser_strict,
+            validation_no_state=validation_no_state,
+            validation_validate_present=validation_validate_present,
+        )
+
+    def parse_data_filepath(
+        self,
+        fileobj: IO,
+        fmt: str,
+        parser_lyb_mod_update: bool = False,
+        parser_no_state: bool = False,
+        parser_parse_only: bool = False,
+        parser_opaq: bool = False,
+        parser_ordered: bool = False,
+        parser_strict: bool = False,
+        validation_no_state: bool = False,
+        validation_validate_present: bool = False,
+    ) -> Optional[DNode]:
+        return self.parse_data(
+            fmt,
+            in_type=IOType.FILEPATH,
+            in_data=fileobj,
```

Comment:
This should call `lyd_parse_data_filepath` directly. I don't see the point in wrapping the call into a generic parse_data that ends up calling specific functions.